<img src="http://rosalind.info/static/img/logo.png?v=1560257990" width="200">

***

**ROSALIND** is a platform for learning bioinformatics and programming through problem solving. Here are my solutions for these problems.

**website:** [http://rosalind.info](http://rosalind.info)

**my accounts:**http://rosalind.info/users/huihui0228/

**problems & solutions:**

* DNA: Counting DNA Nucleotides [[info](http://rosalind.info/problems/dna/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_dna.py)]
* RNA: Transcribing DNA into RNA [[info](http://rosalind.info/problems/rna/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_rna.py)]
* REVC: Complementing a Strand of DNA [[info](http://rosalind.info/problems/revc/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_revc.py)]
* FIB: Rabbits and Recurrence Relations [[info](http://rosalind.info/problems/fib/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_fib.py)]
* GC: Computing GC Content [[info](http://rosalind.info/problems/gc/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_gc.py)]
* HAMM: Counting Point Mutations [[info](http://rosalind.info/problems/hamm/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_hamm.py)]
* IPRB: Mendel's First Law [[info](http://rosalind.info/problems/iprb/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_iprb.py)]
* PROT: Translating RNA into Protein [[info](http://rosalind.info/problems/prot/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_prot.py)]
* SUBS: Finding a Motif in DNA [[info](http://rosalind.info/problems/subs/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_subs.py)]
* CONS: Consensus and Profile [[info](http://rosalind.info/problems/cons/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_cons.py)]
* FIBD: Mortal Fibonacci Rabbits [[info](http://rosalind.info/problems/fibd/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_fibd.py)]
* GRPH: Overlap Graphs [[info](http://rosalind.info/problems/grph/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_grph.py)]
* IEV: Calculating Expected Offspring [[info](http://rosalind.info/problems/iev/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_iev.py)]
* LCSM: Finding a Shared Motif [[info](http://rosalind.info/problems/lcsm/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_lcsm.py)]
* LIA: Independent Alleles [[info](http://rosalind.info/problems/lia/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_lia.py)]
* MPRT: Finding a Protein Motif [[info](http://rosalind.info/problems/mprt/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_mprt.py)]
* MRNA: Inferring mRNA from Protein [[info](http://rosalind.info/problems/mrna/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_mrna.py)]
* ORF: Open Reading Frames [[info](http://rosalind.info/problems/orf/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_orf.py)]
* PERM: Enumerating Gene Orders [[info](http://rosalind.info/problems/perm/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_perm.py)]
* PRTM: Calculating Protein Mass [[info](http://rosalind.info/problems/prtm/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_prtm.py)]
* REVP: Locating Restriction Sites [[info](http://rosalind.info/problems/revp/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_revp.py)]
* SPLC: RNA Splicing [[info](http://rosalind.info/problems/splc/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_splc.py)]
* LEXF: Enumerating k-mers Lexicographically [[info](http://rosalind.info/problems/lexf/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_lexf.py)]
* LGIS: Longest Increasing Subsequence [[info](http://rosalind.info/problems/lgis/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_lgis.py)]
* LONG: Genome Assembly as Shortest Superstring [[info](http://rosalind.info/problems/long/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_long.py)]
* PMCH: Perfect Matchings and RNA Secondary Structures [[info](http://rosalind.info/problems/pmch/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_pmch.py)]
* PPER: Partial Permutations [[info](http://rosalind.info/problems/pper/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_pper.py)]
* PROB: Introduction to Random Strings [[info](http://rosalind.info/problems/prob/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_prob.py)]
* SIGN: Enumerating Oriented Gene Orderings [[info](http://rosalind.info/problems/sign/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_sign.py)]
* SSEQ: Finding a Spliced Motif [[info](http://rosalind.info/problems/sseq/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_sseq.py)]
* TRAN: Transitions and Transversions [[info](http://rosalind.info/problems/tran/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_tran.py)]
* TREE: Completing a Tree [[info](http://rosalind.info/problems/tree/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_tree.py)]
* CORR: Error Correction in Reads [[info](http://rosalind.info/problems/corr/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_corr.py)]
* INOD: Counting Phylogenetic Ancestors [[info](http://rosalind.info/problems/inod/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_inod.py)]
* KMER: k-Mer Composition [[info](http://rosalind.info/problems/kmer/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_kmer.py)]
* KMP: Speeding Up Motif Finding [[info](http://rosalind.info/problems/kmp/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_kmp.py)]
* LEXV: Ordering Strings of Varying Length Lexicographically [[info](http://rosalind.info/problems/lexv/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_lexv.py)]
* MMCH: Maximum Matchings and RNA Secondary Structures [[info](http://rosalind.info/problems/mmch/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_mmch.py)]
* PDST: Creating a Distance Matrix [[info](http://rosalind.info/problems/pdst/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_pdst.py)]
* RSTR: Matching Random Motifs [[info](http://rosalind.info/problems/rstr/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_rstr.py)]
* SSET: Counting Subsets [[info](http://rosalind.info/problems/sset/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_sset.py)]
* EVAL: Expected Number of Restriction Sites [[info](http://rosalind.info/problems/eval/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_eval.py)]
* NWCK: Distances in Trees [[info](http://rosalind.info/problems/nwck/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_nwck.py)]
* SETO: Introduction to Set Operations [[info](http://rosalind.info/problems/seto/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_seto.py)]
* DBRU: Constructing a De Bruijn Graphs [[info](http://rosalind.info/problems/dbru/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_dbru.py)]
* ASPC: Introduction to Alternative Splicing [[info](http://rosalind.info/problems/aspc/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_aspc.py)]
* CAT: Catalan Numbers and RNA Secondary Structures [[info](http://rosalind.info/problems/cat/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_cat.py)]
* MOTZ: Motzkin Numbers and RNA Secondary Structures [[info](http://rosalind.info/problems/motz/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_motz.py)]
* RNAS: Wobble Bonding and RNA Secondary Structures [[info](http://rosalind.info/problems/rnas/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_rnas.py)]
* REAR: Reversal Distance [[info](http://rosalind.info/problems/rear/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_rear.py)]
* SORT: Sorting by Reversals [[info](http://rosalind.info/problems/sort/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_sort.py)]
* SPEC: Inferring Protein from Spectrum [[info](http://rosalind.info/problems/spec/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_spec.py)]
* CONV: Comparing Spectra with the Spectral Convolution [[info](http://rosalind.info/problems/conv/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_conv.py)]
* PRSM: Matching a Spectrum to a Protein [[info](http://rosalind.info/problems/prsm/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_prsm.py)]
* FULL: Inferring Peptide from Full Spectrum [[info](http://rosalind.info/problems/full/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_full.py)]
* SGRA: Using the Spectrum Graph to Infer Peptides [[info](http://rosalind.info/problems/sgra/)] [[code](https://github.com/zonghui0228/Rosalind-Solutions/blob/master/code/rosalind_sgra.py)]
